File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>
#include <stdio.h>
#include <string.h>

bool is_armstrong_number(int candidate) {

  // Convert candidate to str
  char scandidate[12];
  sprintf(scandidate, "%d", candidate);

  int sum = 0;
  for (long unsigned i = 0; i < strlen(scandidate); i++) {
    int x = scandidate[i] - '0'; // Convert char to int
    sum += pow(x, strlen(scandidate));
  }

  return sum == candidate;
}

File: python/armstrong-numbers/armstrong_numbers.py
def lenth_of_num(number):
    return len(str(number))

def make_num_to_arr(number):
    return [int(i) for i in str(number)]

def is_armstrong_number(number):
    x = make_num_to_arr(number)
    y = lenth_of_num(number)
    sum = 0
    for i in range(y):
        sum += x[i]**y
    return sum == number 

